bookstore/client/nb_client.py

```diff
@@ -81,21 +82,21 @@ class NotebookSession:  # (NamedTuple):
     #     id: str #'68d9c58f-c57d-4133-8b41-5ec2731b268d',
     #     path: str #'Untitled38.ipynb',
     #     name: str #'',
-    #     type: str #'notebook',
+    #     kind: str #'notebook',
     #     kernel: KernelInfo
     #     notebook: dict # deprecated API {'path': 'Untitled38.ipynb', 'name': ''}}}
 
-    def __init__(self, *args, path, name, type, kernel, notebook, **kwargs):
+    def __init__(self, *args, path, name, kind, kernel, notebook, **kwargs):
         self.model = {
             "path": path,
             "name": name,
-            "type": type,
+            "kind": kind,
```

Comment:
This no longer is able to accept the literal response from the NotebookSession API… to not have this defeats the purpose of the type class. 
This has the form:
https://github.com/jupyter/notebook/blob/01b084381c367bdca1d58b176428b728a67628ca/notebook/services/api/api.yaml#L761-L778
```
  Session:
    description: A session
    type: object
    properties:
      id:
        type: string
        format: uuid
      path:
        type: string
        description: path to the session
      name:
        type: string
        description: name of the session
      type:
        type: string
        description: session type
      kernel:
        $ref: '#/definitions/Kernel'
```

Reply:
Then maybe we should leave "type" as a key but make the api function signature rename the argument variable. Having a reserved word variable name is problematic in python.

Reply:
The function signature is what needs to have it. I'd be fine making it an attribute that has a different name (it's less ideal & goes against the goals of the API… but possible), but being able to unpack the json response without altering it is the specific intent of this API. 

Also, it's technically not a reserved keyword… just a super commonly used function that is not best practice to overwrite. Keywords: https://docs.python.org/3/reference/lexical_analysis.html#keywords. Otherwise this would be a syntax error.

I don't like doing this either and if it weren't for the explicit desire to be isomorphic to the response object I wouldn't push on this.

Reply:
Just wanted to raise it as it can cause unexpected issues in the future. Not a blocker or required change.